lib/src/merge.dart

```diff
@@ -165,12 +165,12 @@ class _MergeExpanded<T> extends StreamTransformerBase<Stream<T>, T> {
       if (!streams.isBroadcast) {
         controller
           ..onPause = () {
-            for (var subscription in subscriptions) {
+            for (final subscription in subscriptions) {
```

Comment:
Why this change?

Reply:
It matches what I did in https://github.com/dart-lang/stream_transform/pull/92

As I have been rereading this code I found it easier to have a stronger signal for which variables were reassigned so I started replacing more `var` with `final`. It's less of a big deal in this small for loop, but I'm strongly considering turning on the lint to enforce it for this package.